src/config/datadog.rs

```diff
@@ -0,0 +1,117 @@
+use super::{ComponentId, Config, SinkOuter, SourceOuter};
+use crate::{
+    sinks::datadog::metrics::DatadogConfig, sources::internal_metrics::InternalMetricsConfig,
+};
+use serde::{Deserialize, Serialize};
+
+static INTERNAL_METRICS_KEY: &str = "#datadog_internal_metrics";
+static DATADOG_METRICS_KEY: &str = "#datadog_metrics";
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
+#[serde(default, deny_unknown_fields)]
+pub struct Options {
+    #[serde(default = "default_enabled")]
+    pub enabled: bool,
+
+    #[serde(default = "default_api_key")]
+    pub api_key: Option<String>,
+}
+
+impl Default for Options {
+    fn default() -> Self {
+        Self {
+            enabled: default_enabled(),
+            api_key: default_api_key(),
+        }
+    }
+}
+
+/// By default, the Datadog feature is enabled.
+fn default_enabled() -> bool {
+    true
+}
```

Comment:
We did indeed.

Enablement of this feature is based on two pieces of configuration:

1. `enabled: true` (implicit/default)
2. The presence of an `api_key`

Enabled defaulting to `true` poses no security risk, since `api_key` is `None` by default.

Initially, I had considered allowing DATADOG_API_KEY (and DD_API_KEY) from the environment. After a quick discussion on Slack with @binarylogic, we concluded that having an env var triggering enablement could be surprising, so this now needs to be explicitly added to config if we want to leverage environment variables.

i.e. I could see:

```toml
[datadog]
api_key = "${DD_API_KEY}"
```

Reply:
The obvious follow-up question to this is -- why have `enabled: <bool>` at all, if `api_key` alone is sufficient to turn this on/off.

I think it's worth having a separate (effectively hidden) option to allow for configuration that has 'complete' values, but a feature needs to be turned off temporarily for whatever reason.

e.g. I could see:

```toml
[datadog]
enabled = false
api_key = "${DD_API_KEY}"
```

Where the API key is pulled from the environment, but in this particular config, the feature _itself_ is turned off.

This has prior art with the API, where we could do something like:

```toml
[api]
enabled = false
address = "0.0.0.0:${PORT}"
playground = true
```